Constant-related build errors on 32-bit platforms

tracing: 6d8b995c99f1d7b8758da687e5b7df25456f9559
This commit has `tracing-subscriber` depending on `sharded-slab` 0.0.6

When building `tracing-subscriber` for a 32 bit target, e.g. `cargo build --target i686-linux-android`, this results in build errors in `sharded-slab` code.

```
   Compiling sharded-slab v0.0.6
...
   Compiling tracing-subscriber v0.2.0-alpha.1 (/Users/tk/repos/tracing/tracing-subscriber)
error[E0080]: could not evaluate constant
  --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/page/slot.rs:54:24
   |
54 |     const LEN: usize = (cfg::WIDTH - C::RESERVED_BITS) - Self::SHIFT;
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to subtract with overflow

error[E0080]: could not evaluate constant
   --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/lib.rs:691:26
    |
691 |         let shift = 1 << (Self::LEN - 1);
    |                          ^^^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: could not evaluate constant
  --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/cfg.rs:40:30
   |
40 |     const USED_BITS: usize = Generation::<Self>::LEN + Generation::<Self>::SHIFT;
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: could not evaluate constant
   --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/cfg.rs:142:34
    |
142 |             .field("used_bits", &C::USED_BITS)
    |                                  ^^^^^^^^^^^^ referenced constant has errors

error[E0080]: could not evaluate constant
   --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/lib.rs:704:25
    |
704 |     const MASK: usize = Self::BITS << Self::SHIFT;
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: could not evaluate constant
   --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/page/slot.rs:560:24
    |
560 |     const LEN: usize = Generation::<C>::LEN;
    |                        ^^^^^^^^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: could not evaluate constant
   --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/page/slot.rs:495:37
    |
495 |     const LEN: usize = cfg::WIDTH - (Lifecycle::<C>::LEN + Generation::<C>::LEN);
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: could not evaluate constant
   --> /Users/tk/.cargo/registry/src/github.com-1ecc6299db9ec823/sharded-slab-0.0.6/src/lib.rs:699:26
    |
699 |     const SHIFT: usize = Self::Prev::SHIFT + Self::Prev::LEN;
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0080`.
error: could not compile `tracing-subscriber`.
```

--- a/src/cfg.rs
+++ b/src/cfg.rs
@@ -126,7 +126,7 @@ impl Config for DefaultConfig {
     const INITIAL_PAGE_SIZE: usize = 32;
 
     #[cfg(target_pointer_width = "64")]
-    const MAX_THREADS: usize = 4096;
+    const MAX_THREADS: usize = 0xFFF;
     #[cfg(target_pointer_width = "32")]
     const MAX_THREADS: usize = 2048;
 

--- a/src/cfg.rs
+++ b/src/cfg.rs
@@ -39,7 +39,7 @@ pub trait Config: Sized {
 pub(crate) trait CfgPrivate: Config {
     const USED_BITS: usize = Generation::<Self>::LEN + Generation::<Self>::SHIFT;
     const INITIAL_SZ: usize = next_pow2(Self::INITIAL_PAGE_SIZE);
-    const MAX_SHARDS: usize = next_pow2(Self::MAX_THREADS);
+    const MAX_SHARDS: usize = next_pow2(Self::MAX_THREADS - 1);
     const ADDR_INDEX_SHIFT: usize = Self::INITIAL_SZ.trailing_zeros() as usize + 1;
 
     fn page_size(n: usize) -> usize {
@@ -126,9 +126,9 @@ impl Config for DefaultConfig {
     const INITIAL_PAGE_SIZE: usize = 32;
 
     #[cfg(target_pointer_width = "64")]
-    const MAX_THREADS: usize = 0xFFF;
+    const MAX_THREADS: usize = 4096;
     #[cfg(target_pointer_width = "32")]
-    const MAX_THREADS: usize = 2048;
+    const MAX_THREADS: usize = 0x7F; // TODO(eliza):
 
     const MAX_PAGES: usize = WIDTH / 2;
 }

--- a/src/cfg.rs
+++ b/src/cfg.rs
@@ -128,7 +128,8 @@ impl Config for DefaultConfig {
     #[cfg(target_pointer_width = "64")]
     const MAX_THREADS: usize = 4096;
     #[cfg(target_pointer_width = "32")]
-    const MAX_THREADS: usize = 0x7F; // TODO(eliza):
+    // TODO(eliza): can we find enough bits to give 32-bit platforms more threads?
+    const MAX_THREADS: usize = 128;
 
     const MAX_PAGES: usize = WIDTH / 2;
 }

